File: Codeforces/1676H2.py
T = int(input())


def merge(arr, l, r):
    ans = 0
    mid = (l+r) // 2
    i, j = mid-1, r-1
    while i >= l and j >= mid:
        if arr[i] >= arr[j]:
            ans += (j-mid+1)
            i -= 1
        else:
            j -= 1

    i, j = mid-1, r-1
    temp = []
    while i >= l and j >= mid:
        if arr[i] >= arr[j]:
            temp.append(arr[i])
            i -= 1
        else:
            temp.append(arr[j])
            j -= 1
    while i >= l:
        temp.append(arr[i])
        i -= 1
    while j >= mid:
        temp.append(arr[j])
        j -= 1
    i = l
    while len(temp)>0:
        arr[i] = temp.pop()
        i += 1

    return ans


def solve(arr, l, r):  # [l, r)
    ans = 0
    if l+1 == r:
        return ans
    ans += solve(arr, l, (l+r)//2)
    ans += solve(arr, (l+r)//2, r)
    ans += merge(arr, l, r)

    return ans


while T > 0:
    n = int(input())
    arr = list(map(int, input().split()))
    print(solve(arr, 0, n))
    T -= 1

File: Codeforces/1676H2.cpp
#include <bits/stdc++.h>

using namespace std;

int arr[400005];

int solve_segmentTree(int n)
{
    return 0;
}


long long solve_merge(int n)
{
    int len = 1;
    long long ans = 0;
    for (; len <= n; len *= 2)
    {
        for (int l = 1; l <= n; l += 2 * len)
        {
            int i = l + len - 1;
            int j = min(l + 2 * len - 1, n);
            if (i == n)
                continue;
            while (i >= l && j >= l + len)
            {
                if (arr[i] >= arr[j])
                {
                    ans += (long long)(j - l - len) + (long long)1;
                    i--;
                }
                else
                {
                    j--;
                }
            }
            i = l + len - 1;
            j = min(l + 2 * len - 1, n);
            stack<int> temp;
            while (i >= l && j >= l + len)
            {

                if (arr[i] >= arr[j])
                {
                    temp.push(arr[i]);
                    i--;
                }
                else
                {
                    temp.push(arr[j]);
                    j--;
                }
            }
            while (i >= l)
            {
                temp.push(arr[i]);
                i--;
            }
            while (j >= l + len)
            {
                temp.push(arr[j]);
                j--;
            }

            i = l;
            while (!temp.empty())
                arr[i++] = temp.top(), temp.pop();
        }
    }

    return ans;
}

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        int n;
        cin >> n;
        for (int i = 1; i <= n; i++)
        {
            scanf("%d", arr + i);
        }
        long long ans = solve_merge(n);
        cout << ans << '\n';
    }
    return 0;
}